src/engine/MaxConstraint.cpp

```diff
@@ -85,35 +85,39 @@ void MaxConstraint::restoreState( const PiecewiseLinearConstraint *state )
 
 void MaxConstraint::registerAsWatcher( ITableau *tableau )
 {
-    tableau->registerToWatchVariable( this, _f );
     for ( unsigned element : _elements )
-    {
-        if ( element == _f )
-            continue;
         tableau->registerToWatchVariable( this, element );
-    }
+
+    if ( !_elements.exists( _f ) )
+        tableau->registerToWatchVariable( this, _f );
 }
 
 void MaxConstraint::unregisterAsWatcher( ITableau *tableau )
 {
-    tableau->unregisterToWatchVariable( this, _f );
+    // Guy: I think there's a bug here, not related to the recent changes:
+    // the list of _elements can change, and so some of them might not
+    // get propertly unregistered.
```

Comment:
Yes.
When element is removed, it should be unregistered.
But this does not have function which unregister just one element.

Reply:
Okay. Lets ignore this for now, and I'll open an issue so that we can take care of it later.